### ISSUE TITLE
Error: 'ticks' is not exported by node_modules/d3-array/dist/d3-array.js

### ISSUE THREAD

**Issue by devin2712** [sghall/react-compound-slider#25]
### Problem or feature description:

Hitting this error when building with `react-compound-slider` @ `0.15.1`.

```
$ rollup -c

index.ts → dist/index.js...
[!] Error: 'ticks' is not exported by node_modules/d3-array/dist/d3-array.js
https://github.com/rollup/rollup/wiki/Troubleshooting#name-is-not-exported-by-module
node_modules/react-compound-slider/es/Slider/LinearScale.js (3:9)
1: function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }
2:
3: import { ticks } from 'd3-array';
            ^
4:
5: var LinearScale = function () {
Error: 'ticks' is not exported by node_modules/d3-array/dist/d3-array.js
```

### Steps to reproduce (for problems):

Using `0.15.1` (doesn't occur with `0.15.0`), and the following rollup config:
```import commonJS from 'rollup-plugin-commonjs';
import resolve from 'rollup-plugin-node-resolve';
import typescript from 'rollup-plugin-typescript2';
import peerDepsExternal from 'rollup-plugin-peer-deps-external';

const pkg = require('./package.json');

export default {
  input: 'index.ts',

  output: [{ file: pkg.main, format: 'cjs' }],

  watch: { clearScreen: false },
  plugins: [
    peerDepsExternal({ includeDependencies: true }),
    commonJS(),
    typescript({
      exclude: ['storybook/**'],
    }),
    resolve(),
  ],
};
```

### Versions (for problems):

React-Compound-Slider: `0.15.1`

React: `16.5.1`

Browser:

Operating System:

**Comment by devin2712** [comment]
`tsconfig.json` in case that's useful:

{
  "compilerOptions": {
    "target": "es2015",
    "lib": ["dom", "es2015", "es2017"],
    "jsx": "react",
    "declaration": true,
    
    "strict": true,
    "strictNullChecks": true,

    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "noImplicitReturns": true,
    "noFallthroughCasesInSwitch": true, 
    
    "moduleResolution": "node", 
    "allowSyntheticDefaultImports": true,
    "esModuleInterop": true,

    "baseUrl": "./",
    "paths": {
      "react": ["node_modules/@types/react"]
    }
  }
}

**Comment by sghall** [comment]
Hmmm...interesting. Let me take a look.  

Just looking at the source, 'ticks' is exported from 'd3-array'
https://github.com/d3/d3-array/blob/fda02e084d50c880c695a3fe0e59cdf1a1ba5ea9/src/index.js#L24

**Comment by sghall** [comment]
Just looking at this...seems other d3 modules are having this issue...
https://github.com/rollup/rollup/issues/1376

**Comment by sghall** [comment]
Ahhh, looks like you're importing the "dist" version...

```
node_modules/d3-array/dist/d3-array.js
```

 instead of the module version...
https://github.com/d3/d3-array/blob/fda02e084d50c880c695a3fe0e59cdf1a1ba5ea9/package.json#L24

**Comment by sghall** [comment]
Also,  this makes sense that you would encounter this with 0.15.1 because the fix to 0.15.0 was to make the ES module version use ES modules all the way down.   So just need to tweak your rollup config I think.  This comment seems to be on track, but don't have a way to try it out...

https://github.com/rollup/rollup/issues/1376#issuecomment-296359040

Something like:
https://github.com/rollup/rollup-plugin-node-resolve#usage

Make sense?
